api/grpc/mpi/v1/file.proto

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+syntax = "proto3";
+
+package f5.nginx.agent.api.grpc.mpi.v1.file;
+
+import "common.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/nginx/agent/v3/api/grpc/mpi/v1";
+
+// This specifies the FileService operations for transferring file data between a client and server.
+// All operations are written from a client perspective and flow Client -> Server
+// The server must set a max file size (in bytes), and that size must be used to configure 
+// the gRPC server and client for the FileService such that the FileContents object can be sent with bytes of the configured size. 
+// The actual configured max size for gRPC objects must be maxFileSize + sizeOfSha256HashString since a FileContents object contains both. 
+// A SHA256 hash string is 64 bytes, therefore the configured max message size should be maxFileSize + 64.
+service FileService {
+    // Get the overview of files for a particular configuration version of an instance
+    rpc GetOverview(ConfigVersion) returns (FileOverview) {}
+    // Update the overview of files for a particular set of file changes on the data plane
+    rpc UpdateOverview(FileOverview) returns (google.protobuf.Empty) {}
+    // Get the file contents for a particular file
+    rpc GetFile(FileRequest) returns (FileContents) {}
+    // Update a file from the Agent to the Server
+    rpc UpdateFile(File) returns (FileMeta) {}
```

Comment:
Should this be `SendFile`?

Reply:
it can be used for both Add and Updating a file from the server perspective. We don't have a naming convention for this

Reply:
Ok, I saw `SendFile` in the comments, so I was curious which name we are using.

Reply:
I've updated the comment, thanks